site/topics/strings/strings.rst

```diff
@@ -403,99 +403,91 @@ Objects
+* But we have also seen Strings and how they are a little different
+* Strings are *objects* and work a little different
 
 
 Methods
-=======
-* We've seen built in functions 
+-------
+
+* We've seen built in functions
+
     * ``print('this is a function')``
+
 * We've written our own functions
-    * ``char_is_in('a','bleh')``
+
+    * ``character_is_in('a','bleh')``
+
+* However, there is also something called a *method* that is very similar to a function, but acts on a specific instance of an object
+
 
 .. admonition:: Activity
     :class: activity
 
-    In Colab (or whatever IDE):
-        1. Make a string
-        2. Assign it to a variable (if using Colab, hit run too)
-        3. Type the name of the variable
-        4. Press dot (period)
-        5. Wait... (or space or press ctrl-space (depends on IDE))
+    In Colab:
+        #. Assign a string to some variable and print it out
+        #. After running the code, type the name of the variable
+        #. Press ``.`` (period)
+        #. Wait (or press space, or tab, or ctrl-space
+
+    You should see a menu pop up that looks like the following:
 
     .. image:: methods2.png
-    .. image:: methods.png
+
+
+* The menu that popped up contains methods that can be run on a string
+* Other objects that we will see later in the course will have methods associated with them too
 
 .. admonition:: Activity
     :class: activity
 
-    1. Try writing ``a_string.upper()`` and printing it out. 
-    2. Try some other *methods*
-   
-* *Methods* are very very similar to *functions*
-* But we're telling a specific *object* to do something
-* Long story short:
-    * Sometimes we do things with functions
-    * Sometimes we do things with methods
-
-BUT WAIT...
------------
+    #. Assign some string to a variable called ``a_string``
+    #. Add the line of code ``a_string.upper()`` and then ``print`` out ``a_string``
+    #. Try some of the other methods and see what they do
 
-* Why do we have to do it like this ``a_string.upper()``
-* As opposed to like this: ``upper(a_string)``
 
-Answer
-------
+Method vs. Function
+-------------------
 
-1. Because... 
+* Why do we have to do it with a method
 
-2. ``upper(a_string)`` is not actually defined 
+    * ``a_string.upper()``
 
-    * unless we define it ourselves
+* Instead of a function like this?
 
-3. These methods were written by someone, and they wrote them to work a certain way
+    * ``upper(a_string)``
 
-    * Not necessarily the best way, or a way you like
 
-4. There's also a good bookkeeping argument too
+* In reality, there is nothing stopping us from writing the function ``upper(a_string)``
+* But with the string, the sequence of characters that make up the string are stored in the object
 
-    * Put all the string methods with the strings
+    * The method ``a_string.upper()`` is modifying the data within the object
+    * The functionality that modifies the data in that object will belong to that object
```

Comment:
Fun fact, this is actually a bad thing. 

I'll just leave that there.

Reply:
I feel ya --- this is hitting the whole statefull thing. 

Any suggestions on a better presentation?

Reply:
I suppose it could just be _about_ the data in the object.

Reply:
You need to teach objects unfortunately, that's just the state of programming languages. My completely unbacked point is asserting objects are a bad way to encapsulate data and functions -- they are best left to data and functions respectively.